tests/main/auto-refresh-gating/task.yaml

```diff
@@ -164,4 +176,36 @@ execute: |
   echo "Ensure our snap was updated"
   snap list | MATCH "$SNAP_NAME +2\.0\.0"
 
-  # TODO: edge channel, proceed via default behavior. error behavior (hold).
+  systemctl stop snapd.{service,socket}
+
+  # scenario #5
+
+  echo "Checking that error from the hook means hold"
+  echo "--unknown-flag-to-force-snapctl-error" > "$CONTROL_FILE"
+  force_channel_change "$CONTENT_SNAP_NAME" edge
+  force_autorefresh
+
+  systemctl start snapd.{service,socket}
+  LAST_REFRESH_CHANGE_ID=$(wait_for_autorefresh "$CONTENT_SNAP_NAME" "$LAST_REFRESH_CHANGE_ID")
+
+  echo "Ensure our content snap was held"
+  snap list | MATCH "$CONTENT_SNAP_NAME +2\.0\.0"
+
+  snap change --last=auto-refresh | MATCH "ignoring hook error:"
+  snap change --last=auto-refresh | MATCH "ERROR ignoring hook error:"
+  MATCH "error running snapctl: unknown flag .unknown-flag-to-force-snapctl-error'" < "$DEBUG_LOG_FILE"
+
+  systemctl stop snapd.{service,socket}
+
+  # scenario #6
+
+  echo "Checking that if the hook does nothing (neither --proceed nor --hold), the refresh proceeds"
+  rm -f "$CONTROL_FILE"
+  force_channel_change "$CONTENT_SNAP_NAME" edge
+  force_autorefresh
+
+  systemctl start snapd.{service,socket}
+  LAST_REFRESH_CHANGE_ID=$(wait_for_autorefresh "$CONTENT_SNAP_NAME" "$LAST_REFRESH_CHANGE_ID")
```

Comment:
LAST_REFRESH_CHANGE_ID is not used after this, should you check that it is different from the previous one?

Reply:
That is already checked inside wait_for_autorefresh.

Reply:
ah, nice